tests/Unit/SubPageList/Lister/SimpleSubPageFinderTest.php

```diff
@@ -47,5 +62,26 @@ public function testGetSubPagesFor( Title $title ) {
 		$this->assertInternalType( 'array', $pages );
 		$this->assertContainsOnlyInstancesOf( 'Title', $pages );
 	}
+	
+	/**
+	 * @dataProvider redirectProvider
+	 *
+	 * @param Title $title
+	 */
+	public function testGetSubPagesForRedirect( Title $title ) {
+		$finder = $this->newInstance();
+
+		$pages = $finder->getSubPagesFor( $title );
+
+		$this->assertInternalType( 'array', $pages );
+		$this->assertContainsOnlyInstancesOf( 'Title', $pages );
+		
+		foreach ( $pages as $subPage ) {
+			$this->assertEquals( 
```

Comment:
When comparing primitives, `assertSame` is better, as this will also check the type (and not invoke __toString on objects).